connectors/sources/sharepoint.py

```diff
@@ -406,29 +488,28 @@ def format_document(
 
     async def ping(self):
         """Verify the connection with Sharepoint"""
-        if self.session is None:
-            await self._generate_session()
-
-        if self.ssl_enabled and (not self.ssl_ctx):
-            self.ssl_ctx = ssl_context(certificate=self.certificate)
-
         try:
             await anext(
-                self._api_call(
+                self.sharepoint_client.api_call(
```

Comment:
Does this not need to verify the results from the `PING` call at all? I can't see that this checks for a 200 or anything.

Reply:
We are using `raise_for_status` parameter while creating session which throws exception if response status is other than `200`.
that's why we are not verifying the result here